internal/instruction/runner.go

```diff
@@ -289,6 +289,9 @@ func calculateUserInstructionSets(r *Runner, hs b.Homeserver) [][]instruction {
 			instrs = append(instrs, instructionLogin(hs, user))
 		} else {
 			instrs = append(instrs, instructionRegister(hs, user))
+			if user.DisplayName != "" {
+				instrs = append(instrs, instructionDisplayName(hs, user))
+			}
```

Comment:
Am I right in understanding that this is just fixing something that has never worked?

Reply:
Yeah, AFAICS this was meant to be implemented but we forgot.

Reply:
Oh so _that's_ why display names weren't being set correctly. Thanks!